pkg/service/store/store.go

```diff
@@ -165,6 +171,135 @@ func NewClusterService(name, namespace string) ClusterService {
 	}
 }
 
+// MCSAPIServiceSpec is similar as ServiceImportSpec except for the fields: IPs,
+// ExportCreationTimestamp and ServiceName.
+//
+// +k8s:deepcopy-gen=true
+type MCSAPIServiceSpec struct {
+	// ServiceName is the name of the ServiceImport or ServiceExport parent resources.
+	ServiceName string `json:"serviceName"`
+	// ExportCreationTimestamp is a timestamp representing the server time when this object was
+	// created. It is not guaranteed to be set in happens-before order across separate operations.
+	// Clients may not set this value. It is represented in RFC3339 form and is in UTC.
+	//
+	// Populated by the system.
+	// Read-only.
+	// Null for lists.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	ExportCreationTimestamp metav1.Time `json:"exportCreationTimestamp"`
+	// +listType=atomic
+	Ports []mcsapiv1alpha1.ServicePort `json:"ports"`
+	// Type defines the type of this service.
+	// Must be ClusterSetIP or Headless.
+	// +kubebuilder:validation:Enum=ClusterSetIP;Headless
+	Type mcsapiv1alpha1.ServiceImportType `json:"type"`
+	// Supports "ClientIP" and "None". Used to maintain session affinity.
+	// Enable client IP based session affinity.
+	// Must be ClientIP or None.
+	// Defaults to None.
+	// Ignored when type is Headless
+	// More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
+	SessionAffinity v1.ServiceAffinity `json:"sessionAffinity,omitempty"`
+	// SessionAffinityConfig contains session affinity configuration.
+	SessionAffinityConfig *v1.SessionAffinityConfig `json:"sessionAffinityConfig,omitempty"`
+}
+
+// Marshal returns the MCS-API Service Spec object as JSON byte slice
+func (s *MCSAPIServiceSpec) Marshal() ([]byte, error) {
+	return json.Marshal(s)
+}
+
+// Unmarshal parses the JSON byte slice and updates the MCS-API Service Spec receiver
+func (s *MCSAPIServiceSpec) Unmarshal(_ string, data []byte) error {
```

Comment:
What's this `_ string` about?

Reply:
Ah I took it from `ClusterService` which was doing the same thing but it doesn't seems needed indeed